frontend/public/components/namespace-dropdown.tsx

```diff
@@ -0,0 +1,309 @@
+import * as React from 'react';
+import {
+  Button,
+  Divider,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStatePrimary,
+  Menu,
+  MenuContent,
+  MenuGroup,
+  MenuInput,
+  MenuItem,
+  MenuList,
+  Switch,
+  TextInput,
+  Title,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useUserSettingsCompatibility } from '@console/shared';
+import { isSystemNamespace } from './factory/table-filters';
+import NamespaceMenuToggle from './namespace-menu-toggle';
+
+const NamespaceBarDropdown_ = (props: NamespaceBarDropdownProps) => {
+  const { t } = useTranslation();
+  const {
+    actions,
+    disabled,
+    isProjects,
+    onSelect,
+    options,
+    selected,
+    shortCut,
+    storageKey,
+    title,
+    userSettingsPrefix,
+  } = props;
+
+  const MAX_HEIGHT = '60vh';
+
+  const filterRef = React.useRef(null);
+  const menuRef = React.useRef(null);
+
+  const [isMainSelectOpen, setMainSelectOpen] = React.useState(false);
+  const [filterText, setFilterText] = React.useState('');
+  const [activeOption, setActiveOption] = React.useState(selected);
```

Comment:
Is it possible to just use `selected` and not track the active option in state? I'm worried this can get it out sync if we track it in two different places.

Reply:
From an accessibility perspective, the active option and the selected option are actually two different things.  The active item is the item the user last interacted with (usually by using the arrow key).  So when a user leaves the menu (by clicking on tab) and returns to the menu, they should go directly to the active item.  This can (and is often different) than the selected option.  If there isn't an active item set, it should default to the selected item - which is what you see on line 45.

With all that said, remembering the active item doesn't appear to be working as intended. I'll take a look and see if there is an easy way without manually setting tabindex values for this to work correctly.  If not, then I'll just remove the active item logic.